calculator-app/src/test/kotlin/camp/nextstep/edu/calculator/CalculatorViewModelTest.kt

```diff
@@ -124,4 +130,34 @@ class CalculatorViewModelTest {
         // R.string.incomplete_expression == 완성되지 않은 수식입니다.
         assertThat(viewModel.toastEvent.value).isEqualTo(R.string.incomplete_expression)
     }
+
+    // calculate()
+    @Test
+    fun `계산을 하면 로컬 DB에 저장되고 불러오기를 할 수 있다`() {
+        // given : mockk를 활용해 DB를 기능을 구현한다.
+        val calculatorDao: CalculatorDao = mockk()
+        val memoryEntity = MemoryEntity(expression = "33 + 12", result = "45")
+        val memoryEntityList = listOf(memoryEntity)
+
+        every {
+            calculatorDao.insertMemory(memoryEntity)
+        }
+
+        every { calculatorDao.getMemories() } returns memoryEntityList
+
+        val viewModel = CalculatorViewModel(
+            initFormula = listOf<Any>(33, Operator.Plus, 12),
+            calculatorDao = calculatorDao
+        )
+        val expression = "33 + 12"
+        val result = "45"
+
+        // when : 계산을 한다.
+        viewModel.calculate()
+
+
+        // then : DB에 저장된 값과 계산된 결과의 값이 같다.
+        assertThat(calculatorDao.getMemories()[0].expression).isEqualTo(expression)
+        assertThat(calculatorDao.getMemories()[0].result).isEqualTo(result)
```

Comment:
data class이기 때문에 다음과 같이 비교해도 됩니다.
```suggestion
        assertThat(calculatorDao.getMemories()[0]).isEqualTo(memoryEntity)
```